File: src/CorporateMessengerWithAgile/Application/Dto/Summary/UserSummaryDto.cs
namespace Application.Dto.Summary
{
    public class UserSummaryDto : BaseDto
    {
        public string Email { get; set; } = null!;
        public string Username { get; set; } = null!;
        public string PhoneNumber { get; set; } = null!;
        public string Role { get; set; } = null!;
        public IReadOnlyList<Guid> EmployeeIds { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/UserSummaryDto.ts
// src/models/entity/UserSummaryDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const UserSummaryDtoSchema = BaseDtoSchema.extend({
    email: z.email(),
    username: z.string().min(1),
    phoneNumber: z.string().nullable().optional(),
    role: z.string().min(1),
    employeeIds: z.array(GuidSchema).optional(),
});

export type UserSummaryDto = z.infer<typeof UserSummaryDtoSchema>;